docs/guide/using_existing_contracts.rst

```diff
@@ -37,47 +37,47 @@ If you do not have ABI statically, but you know the interface of the contract on
 Fees
 ----
 
-starknet.py requires you to specify amount of Wei you
-are willing to pay either when making ``.invoke()`` transactions or when preparing
-function calls with ``.prepare()``.
+Starknet.py requires you to specify amount of Wei (for V1 transaction) or Fri (for V3 transaction) you
+are willing to pay when executing either ``.invoke_v1()`` or ``.invoke_v3()`` transactions. It's also possible to optionally define the transaction fee
+when preparing function calls with ``.prepare_invoke_v1()`` or ``.prepare_invoke_v3()``.
 
 .. code-block:: python
 
-    await contract.functions["put"].invoke(k, v, max_fee=5000)
+    await contract.functions["put"].invoke_v1(k, v, max_fee=5000)
 
-When max_fee is specified when preparing a call, you can invoke it without
-``max_fee``.
+When max_fee is specified when preparing a call, you can invoke it without ``max_fee``. It works similarly for ``l1_resource_bounds`` and ``prepare_invoke_v3()``.
 
 .. code-block:: python
 
-    prepared_call = contract.function["put"].prepare(k, v, max_fee=5000)
+    prepared_call = contract.function["put"].prepare_invoke_v1(k, v, max_fee=5000)
     await prepared_call.invoke()
 
 .. warning::
 
-    If ``max_fee`` is not specified at any step it will default to ``None``,
-    and will raise an exception when invoking a transaction.
+    For V1 transactions if ``max_fee`` is not specified at any step it will default to ``None``,
+    and will raise an exception when invoking a transaction. The same applies to ``l1_resource_bounds`` and V3 transactions.
```

Comment:
shouldn't we auto-estimate? how does this work, is auto-estimation off by default? or is it exclusive to explicitly setting max_fee to anything?

Reply:
By default, the auto estimation is not enabled. The user must explicitly turn it on by setting `auto_estimate=True`. However, using both `auto_estimate` and `max_fee `simultaneously is not allowed and doing so will cause an error.

I think we shouldn't use auto estimation as a default when `max_fee` is not provided, since it could lead to high and unexpected costs for users. We point out in the documentation that auto estimation is a feature for convenience and should be used mostly for testing.

Reply:
> By default, the auto estimation is not enabled. The user must explicitly turn it on by setting auto_estimate=True. However, using both auto_estimate and max_fee simultaneously is not allowed and doing so will cause an error.

Sure, makes sense

> I think we shouldn't use auto estimation as a default when max_fee is not provided, since it could lead to high and unexpected costs for users. We point out in the documentation that auto estimation is a feature for convenience and should be used mostly for testing.

That's fine as well, but let's make this info more prominent - it should be more discoverable that auto estimation is possible

Reply:
I'm not sure it was addressed, I suggest 
```suggestion
    For V1 transactions if ``max_fee`` is not specified at any step it will default to ``None``,
    and will raise an exception when invoking a transaction, unless `auto_estimate` is specified and is set to `True`. The same applies to ``l1_resource_bounds`` and V3 transactions.
```